src/library_syscall.js

```diff
@@ -717,7 +717,17 @@ var SyscallsLibrary = {
         type = 4; // DT_DIR
       }
       else {
-        var child = FS.lookupNode(stream.node, name);
+        var child;
+        try {
+          child = FS.lookupNode(stream.node, name);
+        } catch (e) {
+          // If the entry is not a directory, file, or symlink, nodefs
+          // lookupNode will raise EINVAL. Skip these and continue.
+          if (e?.errno === {{{ cDefs.EINVAL }}}) {
+            continue;
+          }
+          throw e;
```

Comment:
I wonder if this will interact badly with fseek / fseek if the reported size of the directory doesn't match the number of elements we can read from it?  Is seeking, etc, possible with directories?

Reply:
I think that you only get opaque "cookies" so you can't say "seek forward 5 files" you can just say "where am I right now" and then later seek back to a previous state. For this reason I _think_ this won't cause any harm. It's definitely an improvement over the status quo in any case because currently it stops early.

Reply:
Sounds reasonable yes